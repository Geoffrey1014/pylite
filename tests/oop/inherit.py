class Animal:
    def __init__(self, name, age):
        print "Animal.__init__()"
        self.name = name
        self.age = age

    def get_name(self):
        return self.name

    def get_age(self):
        return self.age

class Human(Animal):
    def __init__(self, name, age, occupation):
        Animal.__init__(self, name, age)
        self.occupation = occupation

    def dump(self):
        print self.get_name()
        print self.get_age()
        print self.occupation

human = Human("mike", 20, "worker")
human.dump()
