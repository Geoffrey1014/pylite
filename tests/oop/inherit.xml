lambda
  name __init__
  param_count 0
  local_count 0
  label_count 0
  insn_vector
    new_class c0
    store_global c0
    load_const c1
    load_global c0
    store_field c2
    load_const c3
    load_global c0
    store_field c4
    load_const c5
    load_global c0
    store_field c6
    new_class c7
    store_global c7
    load_global c0
    load_global c7
    inherit
    load_const c8
    load_global c7
    store_field c2
    load_const c9
    load_global c7
    store_field c10
    load_const c11
    load_const c12
    load_const c13
    load_global c7
    call 3
    store_global c14
    load_global c14
    load_field c10
    call 0
    pop
    load_const c15
    return
  const_vector
    Animal
    lambda
      name __init__
      param_count 3
      local_count 0
      label_count 0
      insn_vector
        load_const c0
        print
        load_local 1
        load_local 0
        store_field c1
        load_local 2
        load_local 0
        store_field c2
        load_const c3
        return
      const_vector
        "Animal.__init__()"
        name
        age
        None
      guard_vector
      line_number_map 3 3 4 4 4 5 5 5 2 2
    __init__
    lambda
      name get_name
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_field c0
        return
      const_vector
        name
      guard_vector
      line_number_map 8 8 8
    get_name
    lambda
      name get_age
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_field c0
        return
      const_vector
        age
      guard_vector
      line_number_map 11 11 11
    get_age
    Human
    lambda
      name __init__
      param_count 4
      local_count 0
      label_count 0
      insn_vector
        load_local 2
        load_local 1
        load_local 0
        load_global c0
        load_field c1
        call 3
        pop
        load_local 3
        load_local 0
        store_field c2
        load_const c3
        return
      const_vector
        Animal
        __init__
        occupation
        None
      guard_vector
      line_number_map 15 15 15 15 15 15 15 16 16 16 14 14
    lambda
      name dump
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_field c0
        call 0
        print
        load_local 0
        load_field c1
        call 0
        print
        load_local 0
        load_field c2
        print
        load_const c3
        return
      const_vector
        get_name
        get_age
        occupation
        None
      guard_vector
      line_number_map 19 19 19 19 20 20 20 20 21 21 21 18 18
    dump
    "worker"
    20.000000
    "mike"
    human
    None
  guard_vector
  line_number_map 1 1 2 2 2 7 7 7 10 10 10 13 13 13 13 13 14 14 14 18 18 18 23 23 23 23 23 23 24 24 24 24 1 1
