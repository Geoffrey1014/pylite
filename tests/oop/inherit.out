Animal.__init__()
mike
20
worker
